fetch.bs

```diff
@@ -3340,88 +3346,104 @@ steps:
+
+ <li><p>Let the <var>revalidatingFlag</var> be unset.
 
  <li>
-  <p>If <var>httpRequest</var>'s
-  <a for=request>cache mode</a> is neither
-  "<code>no-store</code>" nor "<code>reload</code>", and there is a <em>complete</em>
-  <a for=/>response</a> in the HTTP cache for
-  <var>httpRequest</var> run these substeps:
-  <!-- XXX xref "HTTP cache" -->
+  <p>If <var>httpRequest</var>'s <a for=request>cache mode</a> is neither "<code>no-store</code>"
+  nor "<code>reload</code>", run these substeps:
 
   <ol>
    <li>
-    <p>If <var>httpRequest</var>'s <a for=request>cache mode</a> is
-    "<code>force-cache</code>" or "<code>only-if-cached</code>", then set
-    <var>response</var> to the <a for=/>response</a> in the HTTP cache for
-    <var>httpRequest</var>.
+    <p>Let <var>storedResponse</var> be the result of selecting a response from the HTTP cache,
+    possibly needing validation, as per the
+    "<a href=https://tools.ietf.org/html/rfc7234#section-4>Constructing Responses from Caches</a>"
+    chapter of <cite>HTTP Caching</cite> [[!HTTP-CACHING]], if any.
 
-    <p class=note>As mandated by HTTP, this still takes the `<code>Vary</code>`
-    <a>header</a> into account.
+   <li><p>If <var>storedResponse</var> is null, then abort these substeps.
 
-   <li><p>Otherwise, if <var>httpRequest</var>'s
-   <a for=request>cache mode</a> is "<code>default</code>" and the
-   <a for=/>response</a> in the HTTP cache for <var>httpRequest</var> does
-   not require revalidation, then set <var>response</var> to that
-   <a for=/>response</a>.
-   <!-- XXX xref "revalidation" -->
+   <!-- cache hit -->
+   <li><p>If <var>storedResponse</var> requires validation (i.e., it is not fresh), then set the
+   <var>revalidatingFlag</var>.
 
-   <li><p>Otherwise, if <var>httpRequest</var>'s
-   <a for=request>cache mode</a> is either "<code>default</code>"
-   or "<code>no-cache</code>", modify <var>httpRequest</var>'s
-   <a for=request>header list</a> with revalidation
-   <a>headers</a>.
-   <!-- XXX modify, revalidation headers -->
-  </ol>
+   <li>
+    <p>If <var>httpRequest</var>'s <a for=request>cache mode</a> is "<code>force-cache</code>" or
+    "<code>only-if-cached</code>", then set <var>response</var> to <var>storedResponse</var> and
+    abort these substeps.
 
- <li><p>Otherwise, if <var>httpRequest</var>'s
- <a for=request>cache mode</a> is either
- "<code>default</code>" or "<code>force-cache</code>", and there is a
- <em>partial</em> <a for=/>response</a> in the HTTP cache for
- <var>httpRequest</var>, modify <var>httpRequest</var>'s
- <a for=request>header list</a> with resume
- <a>headers</a>.
- <!-- XXX xref partial, modify, resume headers -->
+    <p class=note>As mandated by HTTP, this still takes the `<code>Vary</code>` <a>header</a>
+    into account.
 
- <li>
-  <p>If <var>response</var> is null, run these substeps:
+   <li>
+    <p>If the <var>revalidatingFlag</var> is set, then:
```

Comment:
If I'm reading this correctly, nothing assigns `storedResponse` to `response` if revalidation is not needed, and the cache-mode is not `force-cache` or `only-if-cached`.
Am I reading this correctly? If so, is this intentional? I'd expect such assignment to happen if no revalidation is needed.